src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -207,63 +207,127 @@ public Task<Either<ActionResult, DeleteReleasePlanViewModel>> GetDeleteReleaseVe
-                        .UserReleaseRoles
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    roles.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(roles);
-
-                    var invites = await _context
-                        .UserReleaseInvites
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    invites.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(invites);
-
-                    var methodologiesScheduledWithRelease =
-                        GetMethodologiesScheduledWithRelease(releaseVersionId);
-
-                    // TODO EES-2747 - this should be looked at to see how best to reuse similar "set to draft" logic
-                    // in MethodologyApprovalService.
-                    methodologiesScheduledWithRelease.ForEach(m =>
+                    if (!releaseVersion.Amendment && releaseVersion.ApprovalStatus == ReleaseApprovalStatus.Draft)
                     {
-                        m.PublishingStrategy = Immediately;
-                        m.Status = Draft;
-                        m.ScheduledWithReleaseVersion = null;
-                        m.ScheduledWithReleaseVersionId = null;
-                        m.Updated = DateTime.UtcNow;
-                    });
+                        await HardDeleteForDraft(releaseVersion, cancellationToken);
+                    }
+                    else
+                    {
+                        await SoftDeleteForAmendment(releaseVersion, cancellationToken);
+                    }
 
-                    _context.UpdateRange(methodologiesScheduledWithRelease);
+                    UpdateMethodologies(releaseVersionId);
 
                     await _context.SaveChangesAsync();
 
                     await _releaseSubjectRepository.DeleteAllReleaseSubjects(releaseVersionId: releaseVersionId);
                 });
         }
 
+        private async Task HardDeleteForDraft(
+            ReleaseVersion releaseVersion,
+            CancellationToken cancellationToken)
+        {
+            var publication = await _context.Publications.FindAsync(releaseVersion.PublicationId, cancellationToken);
+            var releaseSeriesItem = publication!.ReleaseSeries.Find(rs => rs.ReleaseId == releaseVersion.ReleaseId);
+
+            publication.ReleaseSeries.Remove(releaseSeriesItem!);
+            _context.Publications.Update(publication);
+
+            var release = await _context.Releases.FindAsync(releaseVersion.ReleaseId, cancellationToken);
+            _context.Releases.Remove(release!);
+
+            await DeleteRoles(releaseVersion.Id, hardDelete: true, cancellationToken);
+            await DeleteInvites(releaseVersion.Id, hardDelete: true, cancellationToken);
```

Comment:
`UserReleaseRoles` doesn't appear to delete by cascade, as removing these failed a test, so I'll leave as-is for now. It's possible `UserReleaseInvites` aren't removed either, but the test fails before reaching that point - I'll leave a comment suggesting that they should be, and to be investigated.

Reply:
If the unit tests failed, then that is probably because the in-memory db isn't doing the cascade delete - while on an actual database the cascade delete would occur. This might actually explain why these lines are here.

Maybe just add a comment saying something like "We suspect this is only necessary for the unit tests - the in-memory db doesn't do the cascade delete".